Sources/SE0000_KeyPathReflection/KeyPathInitialization.swift

```diff
@@ -0,0 +1,147 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import KeyPathReflection_CShims
+
+// This is a utility within KeyPath.swift in the standard library. If this
+// gets moved into there, then this goes away, but will have to rethink if this
+// goes into a different module.
+extension AnyKeyPath {
+  internal static func _create(
+    capacityInBytes bytes: Int,
+    initializedBy body: (UnsafeMutableRawBufferPointer) -> ()
+  ) -> Self {
+    assert(bytes > 0 && bytes % 4 == 0,
+           "capacity must be multiple of 4 bytes")
+    let metadata = getMetadata(for: self) as! ClassMetadata
+    var size = metadata.instanceSize
+    
+    let tailStride = MemoryLayout<Int32>.stride
+    let tailAlignMask = MemoryLayout<Int32>.alignment - 1
+    
+    size += tailAlignMask
+    size &= ~tailAlignMask
+    size += tailStride * (bytes / 4)
+    
+    let alignment = metadata.instanceAlignMask | tailAlignMask
+    
+    let object = swift_allocObject(
+      UnsafeMutableRawPointer(mutating: metadata.pointer),
+      size,
+      alignment
+    )
+    
+    guard object != nil else {
+      fatalError("Allocating \(self) instance failed for keypath reflection")
+    }
+    
+    // This memory layout of Int by 2 is the size of a heap object which object
+    // points to. Tail members appear immediately afterwards.
+    let base = object! + MemoryLayout<Int>.size * 2
+    
+    // The first word is the kvc string pointer. Set it to 0 (nil).
+    base.storeBytes(of: 0, as: Int.self)
+    
+    // Return an offseted base after the kvc string pointer.
+    let newBase = base + MemoryLayout<Int>.size
+    let newBytes = bytes - MemoryLayout<Int>.size
+    
+    body(UnsafeMutableRawBufferPointer(start: newBase, count: newBytes))
+    
+    return unsafeBitCast(object, to: self)
+  }
+}
```

Comment:
The layout of AnyKeyPath is not ABI-stable. This is not guaranteed to work.

Reply:
The idea behind this is simply to have something separate and usable before baking in the standard library. This already exists there (so if it goes in the stdlib, this for away entirely), so before committing to the Swift project I had to reimplement this to have something usable.